app/models/manageiq/providers/hawkular/concerns/server_operations.rb

```diff
@@ -0,0 +1,282 @@
+require 'op_dispatcher'
+
+module ManageIQ::Providers
+  module Hawkular
+    module ServerOperations
+      # server ops
+      def shutdown_middleware_server(ems_ref, params = {})
+        run_generic_operation(:Shutdown, ems_ref, :restart => false, :timeout => params[:timeout] || 0)
+      end
+
+      def suspend_middleware_server(ems_ref, params = {})
+        run_generic_operation(:Suspend, ems_ref, :timeout => params[:timeout] || 0)
+      end
+
+      def resume_middleware_server(ems_ref)
+        run_generic_operation(:Resume, ems_ref)
+      end
+
+      def reload_middleware_server(ems_ref)
+        run_generic_operation(:Reload, ems_ref)
+      end
+
+      def stop_middleware_server(ems_ref)
+        run_generic_operation(:Shutdown, ems_ref)
+      end
+
+      def start_middleware_domain_server(ems_ref)
+        run_generic_operation(:Start, ems_ref)
+      end
+
+      def stop_middleware_domain_server(ems_ref)
+        run_generic_operation(:Stop, ems_ref)
+      end
+
+      def restart_middleware_server(ems_ref)
+        run_generic_operation(:Shutdown, ems_ref, :restart => true)
+      end
+
+      # domain server ops
+      def restart_middleware_domain_server(ems_ref)
+        run_generic_operation(:Restart, ems_ref)
+      end
+
+      def kill_middleware_domain_server(ems_ref)
+        run_generic_operation(:Kill, ems_ref)
+      end
+
+      # server group ops
+      def start_middleware_server_group(ems_ref)
+        run_generic_operation('Start Servers', ems_ref)
+      end
+
+      def stop_middleware_server_group(ems_ref, params = {})
+        run_generic_operation('Stop Servers', ems_ref, :timeout => params[:timeout] || 0)
+      end
+
+      def restart_middleware_server_group(ems_ref)
+        run_generic_operation('Restart Servers', ems_ref)
+      end
+
+      def reload_middleware_server_group(ems_ref)
+        run_generic_operation('Reload Servers', ems_ref)
+      end
+
+      def suspend_middleware_server_group(ems_ref, params = {})
+        run_generic_operation('Suspend Servers', ems_ref, :timeout => params[:timeout] || 0)
+      end
+
+      def resume_middleware_server_group(ems_ref)
+        run_generic_operation('Resume Servers', ems_ref)
+      end
+
+      def create_jdr_report(ems_ref)
+        run_generic_operation(:JDR, ems_ref)
```

Comment:
afaik, JDR reports are blocked by a flaw in the websocket library we are using. It was very likely it fails with seg fault when receiving the binary websocket frame. The probability was higher when the data was larger, so with JDR report it was almost sure, it fails. Once Ruben's [work](https://github.com/hawkular/hawkular-client-ruby/pull/205) is merged, this can be added.